src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -395,6 +406,11 @@ MulticopterPositionControl::poll_subscriptions()
 				_attitude_setpoint_id = ORB_ID(vehicle_attitude_setpoint);
 			}
 		}
+
+		// if vehicle is a VTOL we want to enable weathervane capabilities
+		if (_wv_controller == nullptr && _vehicle_status.is_vtol) {
+			_wv_controller = new WeatherVane();
```

Comment:
Why do we now start with dynamic allocation?

Reply:
@Stifael Because we don't want to allocate memory for vehicles that don't need this.

Reply:
It also prevents the weathervane (WV_*) parameters from being marked active and synced to the ground for display. Regular MC users don't have to carry the burden in any way.

Reply:
I am just worried that this will become a common implementation strategy. For instance, from an implementation point of view weathervane and avoidance are similar because also avoidance right now is only used for MC. In addition, I am sure there will be other special controllers in the future. I fear that dynamic allocation will become the general way of adding new special controllers.

Reply:
Dynamic allocation is perfectly fine if done once initially. That being said, there are ideas being pursued (eg https://github.com/PX4/Firmware/issues/8814) that would significantly reduce the overhead per module and then allow us to think about structuring these things differently.